crates/libcontainer/src/process/container_init_process.rs

```diff
@@ -591,7 +581,15 @@ pub fn container_init_process(
         tracing::warn!("seccomp not available, unable to set seccomp privileges!")
     }
 
+    // add HOME into envs if not exists
+    if !envs.contains_key("HOME") {
+        if let Some(dir_home) = utils::get_user_home(proc.user().uid()) {
+            envs.insert("HOME".to_owned(), dir_home.to_string_lossy().to_string());
+        }
+    }
+
     args.executor.validate(spec)?;
+    args.executor.set_envs(envs)?;
```

Comment:
Is the idea for the wasm shim to override this method by giving an empty implementation and decides what to do with the env vars inside of the `Executor::execute()` function?

Reply:
> Is the idea for the wasm shim to override this method by giving an empty implementation and decides what to do with the env vars inside of the Executor::execute() function?

Yes, that's my intention.

Reply:
Another option was moving the logic of setting envs under the `Executor::execute()` default implementation, which would affect any existing `Executor` implementations that assume the original behavior regarding environment variables. IMO defining another dedicated method would be good as suggested in https://github.com/containers/youki/issues/2815#issuecomment-2169088272.